File: JAVA/713. Subarray Product Less Than K.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if(k == 0) return 0;
        int res = 0;
        int product = 1;
        int l = 0;
        for(int r = 0; r < nums.length; r++) {
            if(nums[r] >= k) {
                l = r + 1;
                product = 1;
                continue;
            }
            product *= nums[r];
            while (product >= k) {
                product /= nums[l++];
            }
            res += r - l + 1;
        }
        return res;
    }
}

File: C++/713. Subarray Product Less Than K.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {

        int res = 0;
        if(k == 0){
            return res;
        }
        for(int i = 0 ; i < nums.size() ; i++){
            long long temp = 1;
            for(int j = i ; j < nums.size() ; j++){
                temp = nums[j]*temp;
                if(temp<k){
                    res++;
                }else{
                    break;
                }
            }
        }
        return res;
    }
};
